types/meteor/ejson.d.ts

```diff
@@ -1,53 +1,46 @@
 declare module 'meteor/ejson' {
     interface EJSONableCustomType {
         clone?(): EJSONableCustomType;
-        equals?(other: Object): boolean;
+        equals?(other: EJSONable): boolean;
         toJSONValue(): JSONable;
         typeName(): string;
     }
-    type EJSONableProperty =
+    export type EJSONable =
         | number
         | string
         | boolean
-        | Object
-        | number[]
-        | string[]
-        | Object[]
         | Date
         | Uint8Array
         | EJSONableCustomType
         | undefined
-        | null;
-    interface EJSONable {
-        [key: string]: EJSONableProperty;
-    }
-    interface JSONable {
-        [key: string]: number | string | boolean | Object | number[] | string[] | Object[] | undefined | null;
-    }
-    interface EJSON extends EJSONable {}
+        | null
+        | EJSONable[]
+        | { [key: string]: EJSONable };
+
+    type JSONable = string | number | boolean | null | JSONable[] | { [key: string]: JSONable };
```

Comment:
I'm not sure if that's a correct change (see https://github.com/microsoft/TypeScript/issues/1897 for details). Long story short, it doesn't handle custom serializers.

(Similarly, `fromJSONValue` has to return `any` _or_ have a type parameter to handle that case.)

I wouldn't touch these types just yet in this PR.

Reply:
That's reasonable. What do you recommend using instead of `Object`, then? (Like, `Object` is generic enough that we might as well just call it `any`)

Reply:
There's nothing really. I'd just leave it for now as `any` is far more permissive.